### PR TITLE
Dump log in outputDir found in config and add an clean error if the n…

### PR DESCRIPTION
**Pull request by rosiereflo**
…etwork file is not found

### CODE REVIEW

**Comment by rosiereflo** [comment]
#32

File: sources/Common/src/Log.cpp [hunk 1]
```diff
@@ -15,20 +15,28 @@
...
 void
-Log::init(const common::Options& options) {
+Log::init(const common::Options& options, const std::string& outputDir) {
   using DYN::Trace;
   auto& config = options.config();
 
   std::vector<Trace::TraceAppender> appenders;
   Trace::TraceAppender appender;
+  file::path outputPath(outputDir);
+  if (!file::exists(outputPath)) {
```
Comment (lecourtoisflo) [inline sources/Common/src/Log.cpp:35]: maybe remove the corresponding lines in context.exportOuputs

File: sources/Context.h [hunk 2]
```diff
@@ -118,7 +118,7 @@ class Context {
  private:
   ContextDef def_;                         ///< context definition
   inputs::NetworkManager networkManager_;  ///< network manager
-  inputs::Configuration config_;           ///< configuration
+  const inputs::Configuration& config_;           ///< configuration
```
Comment (lecourtoisflo) [inline sources/Context.h:121]: incorrect formatting

File: sources/main.cpp [hunk 3]
```diff
@@ -76,17 +79,21 @@ main(int argc, char* argv[]) {
     if (!boost::filesystem::is_regular_file(dictPath)) {
       // we cannot use dictionnary errors since they are not be initialized yet
       LOG(error) << "Dictionary " << dictPath << " not found: check runtime environment" << LOG_ENDL;
+      std::cerr << "Dictionary " << dictPath << " not found: check runtime environment" << std::endl;
```
Comment (lecourtoisflo) [inline sources/main.cpp:82]: why use cerr ? Don't we want to use the log instead ? cerr should be used only when it is unsure that log file is initialized

File: sources/main.cpp [hunk 4]
```diff
@@ -76,17 +79,21 @@ main(int argc, char* argv[]) {
...
       return EXIT_FAILURE;
     }
-
-    auto& config = options.config();
-    LOG(info) << MESS(InputsInfo, config.networkFilePath, config.configPath) << LOG_ENDL;
+    if (!boost::filesystem::exists(boost::filesystem::path(runtimeConfig.networkFilePath))) {
+      LOG(error) << MESS(NetworkFileNotFound, runtimeConfig.networkFilePath) << LOG_ENDL;
+      std::cerr << MESS(NetworkFileNotFound, runtimeConfig.networkFilePath) << std::endl;
```
Comment (lecourtoisflo) [inline sources/main.cpp:87]: same here as previous remark about cerr

File: sources/Context.h [hunk 5]
```diff
@@ -53,8 +52,9 @@ class Context {
    * @brief Constructor
    *
    * @param def The context definition
+   * @param config simulation configuration
```
Comment (lecourtoisflo) [inline sources/Context.h:55]: it is not simulation configuration rather than DFL specific configuration

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]

File: sources/main.cpp (hunk 3 above)
Comment (rosiereflo) [reply sources/main.cpp:82]: It find it nicer to have a feedback in the terminal when an error happened, but that's only personal
@marcochiaramello an opinion on this?

Comment (marcochiaramello) [reply sources/main.cpp:82]: I do agree, maybe in debug mode if possible..otherwise I'll keep it

File: sources/Common/src/Log.cpp (hunk 1 above)
Comment (rosiereflo) [reply sources/Common/src/Log.cpp:35]: done

File: sources/Context.h (hunk 5 above)
Comment (rosiereflo) [reply sources/Context.h]: done

File: sources/Context.h (hunk 2 above)
Comment (rosiereflo) [reply sources/Context.h]: done

File: sources/main.cpp (hunk 3 above)
Comment (lecourtoisflo) [reply sources/main.cpp:82]: In that case, it should be done in dynawo, in function error from Trace

Comment (rosiereflo) [reply sources/main.cpp]: ok I removed them

File: sources/Context.h [hunk 6]
```diff
@@ -53,8 +52,9 @@ class Context {
    * @brief Constructor
    *
    * @param def The context definition
+   * @param config configuration to use
    */
-  explicit Context(const ContextDef& def);
+  explicit Context(const ContextDef& def, const inputs::Configuration& config);
```
Comment (lecourtoisflo) [inline sources/Context.h:57]: explicit is no longer necessary

File: sources/Common/include/Log.h [hunk 7]
```diff
@@ -36,8 +36,9 @@ class Log {
    * - one logger with the name of the program, containing logs from dynaflow launcher
    * - one logger named dynawo.log, containing the rest of dynawo logs
```
Comment (lecourtoisflo) [inline sources/Common/include/Log.h:37]: comment not right any more. Also missing comment for parameter options from previous develpments

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]

File: sources/Context.h (hunk 6 above)
Comment (rosiereflo) [reply sources/Context.h]: done

File: sources/Common/include/Log.h (hunk 7 above)
Comment (rosiereflo) [reply sources/Common/include/Log.h]: done

**Review by lecourtoisflo — APPROVED** [review]

**Pull request merged**